src/core/geometry/geom3/measureVolume.js

```diff
@@ -1,9 +1,13 @@
 const poly3 = require('../poly3')
-const toTriangles = require('./toTriangles')
 
-const measureVolume = geometry => {
-  let result = toTriangles(geometry).map(triPoly => poly3.getSignedVolume(triPoly))
-  console.log('volume', result)
+/*
+ * Measure the volume of the given geometry.
+ *
+ * @param {Geom3} geometry - 3D geometry to measure
+ * @returns {Number} volume of the geometry
+ */
+const measureVolume = (geometry) => {
+  let result = geometry.polygons.reduce((volume, polygon) => volume + poly3.measureSignedVolume(polygon), 0)
```

Comment:
why the signed volume?

Reply:
I tried to test volumes in ’negative’ space, but found negative - negative produced positive volumes.

Any thoughts?

Reply:
good point, is a volume not always positive ? or does the sign have a significance ?

Reply:
It seems to be positive always. The research paper seems to suggest that negative volumes are possible, but I cannot produce a test case.